pvextractor/geometry/path.py

```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
         return polygons
 
     def to_patches(self, spacing, wcs=None, **kwargs):
-        from matplotlib.patches import Polygon as MPLPolygon
-        patches = []
-        for poly in self.sample_polygons(spacing, wcs=wcs):
-            patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
-        return patches
+        if self.width is not None:
+            from matplotlib.patches import Polygon as MPLPolygon
+            patches = []
+            for poly in self.sample_polygons(spacing, wcs=wcs):
+                patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
+            return patches
+        else:
+            raise ValueError("Use as_artist instead")
+
+    def as_artist(self, spacing, wcs=None, facecolor='none', **kwargs):
+        """
+        Display the path on the image
+
+        Parameters
+        ----------
+        spacing : float
+            The spacing between sample points to plot, in pixels
+        wcs : astropy.wcs.WCS
+            A WCS instance to convert from celestial to pixel coordinates, or
+            None if pixel coordinates are to be used.
+        facecolor : string
+            The facecolor of the patches.  This will be ignored if lines are
+            plotted and defaults to none (transparent) for patches.
+        """
+        if self.width is None:
+            from matplotlib.lines import Line2D
+            points = self.sample_points(spacing, wcs=wcs)
+            artist = Line2D(*points, **kwargs)
+        else:
+            from matplotlib.collections import PatchCollection
+            patches = self.to_patches(spacing, wcs=wcs, facecolor=facecolor,
+                                      **kwargs)
+            artist = PatchCollection(patches, facecolor=facecolor, **kwargs)
+        return artist
+
+    def show_on_axis(self, ax, spacing, **kwargs):
+        """
+        Add the Path as a set of patches on the specified axis
+
+        Parameters
+        ----------
+        ax : matplotlib.axes._subplots.WCSAxesSubplot
+            An astropy WCSAxesSubplot axis to overplot onto
+        spacing : float
+            The spacing between sample points to plot, in pixels
+        """
+        artist = self.as_artist(spacing=spacing,
+                                wcs=ax.wcs,
+                                **kwargs)
+        ax.add_artist(artist)
+        return artist
```

Comment:
Does `artist` need to be returned? `as_artist` could be used instead if only the artist object is needed.

Reply:
Doesn't need to, but matplotlib functions all return the artists, so I think we should

Reply:
👍